streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamWindowAggregateTest.java

```diff
@@ -315,59 +310,103 @@ public void shouldLogAndMeterWhenSkippingExpiredWindowByGrace() {
         final String topic = "topic";
 
         final KStream<String, String> stream1 = builder.stream(topic, Consumed.with(Serdes.String(), Serdes.String()));
-        stream1.groupByKey(Serialized.with(Serdes.String(), Serdes.String()))
-            .windowedBy(TimeWindows.of(10).advanceBy(5).grace(90L))
-            .aggregate(
-                () -> "",
-                MockAggregator.toStringInstance("+"),
-                Materialized.<String, String, WindowStore<Bytes, byte[]>>as("topic1-Canonicalized").withValueSerde(Serdes.String()).withCachingDisabled().withLoggingDisabled()
-            )
-            .toStream()
-            .map((key, value) -> new KeyValue<>(key.toString(), value))
-            .to("output");
+        stream1.groupByKey(Grouped.with(Serdes.String(), Serdes.String()))
+               .windowedBy(TimeWindows.of(10).advanceBy(10).grace(90L))
+               .aggregate(
+                   () -> "",
+                   MockAggregator.toStringInstance("+"),
+                   Materialized.<String, String, WindowStore<Bytes, byte[]>>as("topic1-Canonicalized").withValueSerde(Serdes.String()).withCachingDisabled().withLoggingDisabled()
+               )
+               .toStream()
+               .map((key, value) -> new KeyValue<>(key.toString(), value))
+               .to("output");
 
         LogCaptureAppender.setClassLoggerToDebug(KStreamWindowAggregate.class);
         final LogCaptureAppender appender = LogCaptureAppender.createAndRegister();
         try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), props, 0L)) {
-            driver.pipeInput(recordFactory.create(topic, "k", "100", 100L));
-            driver.pipeInput(recordFactory.create(topic, "k", "0", 0L));
-            driver.pipeInput(recordFactory.create(topic, "k", "1", 1L));
-            driver.pipeInput(recordFactory.create(topic, "k", "2", 2L));
-            driver.pipeInput(recordFactory.create(topic, "k", "3", 3L));
-            driver.pipeInput(recordFactory.create(topic, "k", "4", 4L));
-            driver.pipeInput(recordFactory.create(topic, "k", "5", 5L));
+            driver.pipeInput(recordFactory.create(topic, "k", "100", 200L));
+            driver.pipeInput(recordFactory.create(topic, "k", "0", 100L));
+            driver.pipeInput(recordFactory.create(topic, "k", "1", 101L));
+            driver.pipeInput(recordFactory.create(topic, "k", "2", 102L));
+            driver.pipeInput(recordFactory.create(topic, "k", "3", 103L));
+            driver.pipeInput(recordFactory.create(topic, "k", "4", 104L));
+            driver.pipeInput(recordFactory.create(topic, "k", "5", 105L));
             driver.pipeInput(recordFactory.create(topic, "k", "6", 6L));
```

Comment:
With the prior test data, all the records would have the exact same amount of lateness. I created a difference, which lets us demonstrate correct max and avg lateness metrics.